src/jitlayers.cpp

```diff
@@ -497,7 +503,36 @@ static void addPassesForOptLevel(legacy::PassManager &PM, TargetMachine &TM, raw
 
 CompilerResultT JuliaOJIT::CompilerT::operator()(Module &M)
 {
+    uint64_t start_time = 0;
+    if (dump_llvm_opt_stream != NULL) {
+        // Print LLVM function statistics _before_ optimization
+        // Print all the information about this invocation as a YAML object
+        jl_printf(dump_llvm_opt_stream, "- \n");
+        // We print the name and some statistics for each function in the module, both
+        // before optimization and again afterwards.
+        jl_printf(dump_llvm_opt_stream, "  before: \n");
+        for (auto &F : M.functions()) {
+            if (F.getInstructionCount() == 0 || F.getName().str().rfind("jfptr_", 0) == 0) {
```

Comment:
```suggestion
            if (F.isDeclaration()) {
```

I'm not sure why the other check matters / makes sense (but seems like it should at least use [`startsWith`](https://llvm.org/doxygen/classllvm_1_1StringRef.html#aeb540aeadd339af860f5c217a85363b7))?

Reply:
We were leaving off the `jfptr_` functions, because they seem to be auto-generated boxing/unboxing functions, and just add noise to the output. I don't really remember exactly why we left them out... 

I can experiment with putting them back in, and see what happens? (We've been so focused on inference times the last three weeks, I've forgotten what some of this code does already)

Reply:
> We were leaving off the jfptr_ functions, because they seem to be auto-generated boxing/unboxing functions, and just add noise to the output. I don't really remember exactly why we left them out...

My memory of the `F.getName().str().rfind("jfptr_", 0)` check is hazy by this point as well, but what memory I do have is consistent with your recollection, i.e. the `jfptr_`-related output practically seemed like noise.

The `isDeclaration` and `startsWith` rewrites sound great! :)

Reply:
Done. I've rewritten this as `if (F.isDeclaration() || F.getName().startswith("jfptr_"))`.

Reply:
And after reflecting over this a bit, i think we skipped them simply because they don't add anything to the user.

The question one wants to answer is how much LLVM time did each julia function contribute to overall compilation times. we don't have any julia function to associate with the `jfptr_*` functions, so they will remain C function names, and from what i understand, every function will get on of them generated for it, and they will all be equally expensive to compile. So i don't think reporting them adds anything. They'll contribute noise in the downstream analyses, so I'm in favor of continuing to leave them off. :)

Does that make sense? Thanks!